spec/job/adapters/kubernetes/batch_spec.rb

```diff
@@ -414,6 +417,71 @@ def build_script(opts = {})
       @basic_batch.submit(script)
     end
 
+    it "submits with correct yml file given extra supplemental groups" do
+      script = build_script(
+        accounting_id: 'test',
+        content: script_content,
+        native: {
+          container: {
+            name: 'rspec-test',
+            image: 'ruby:2.5',
+            command: 'rake spec',
+            port: 8080,
+            env: {
+              'HOME' => '/my/home',
+              'PATH' => '/usr/bin:/usr/local/bin'
+            },
+            memory: '6Gi',
+            cpu: '4',
+            working_dir: '/my/home',
+            restart_policy: 'Always',
+            supplemental_groups: [1002],
+          },
+          init_containers: [
+            name: 'init-1',
+            image: 'busybox:latest',
+            command: '/bin/ls -lrt .'
+          ],
+          configmap: {
+            files: [{
+              filename: 'config.file',
+              data: "a = b\nc = d\n  indentation = keepthis",
+              mount_path: '/ood'
+            }],
+          },
+          mounts: [
+            type: 'host',
+            name: 'ess',
+            host_type: 'Directory',
+            destination_path: '/fs/ess',
+            path: '/fs/ess'
+          ]
+        }
+      )
+
+      allow(@basic_batch).to receive(:generate_id).with('rspec-test').and_return('rspec-test-123')
+      allow(@basic_batch).to receive(:username).and_return('testuser')
+      allow(@basic_batch).to receive(:user).and_return(User.new(dir: '/home/testuser', uid: 1001, gid: 1002))
+      allow(@basic_batch).to receive(:group).and_return('testgroup')
+      allow(@basic_batch).to receive(:enable_supplemental_groups).and_return(true)
+      allow(@basic_batch).to receive(:default_supplemental_groups).and_return([1000,1001])
+
+      # make sure it get's templated right, also helpful in debugging bc
+      # it'll show a better diff than the test below.
+      template, = @basic_batch.send(:generate_id_yml, script)
+      expect(template.to_s).to eql(pod_yml_extra_groups.to_s)
+
+      # make sure template get's passed into command correctly
+      allow(Open3).to receive(:capture3).with(
+        {},
+        "/usr/bin/kubectl --kubeconfig=#{ENV['HOME']}/.kube/config " \
+        "--namespace=testuser -o json create -f -",
+        stdin_data: pod_yml_extra_groups.to_s
+      ).and_return(['', '', success])
+
+      @basic_batch.submit(script)
+    end
```

Comment:
This is what I meant by the need to refactoring here. There's a whole lot of test case here and what we're testing is a very slight difference from all the other test cases.

Reply:
Nothing to do here on this PR - just noting it for #306.